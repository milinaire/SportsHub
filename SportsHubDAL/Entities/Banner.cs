using SportsHubDAL.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SportsHubDAL.Entities
{
    public class Banner : IDBEntity
    {
        public int Id { get; set; }
        public bool IsPublished { get; set; }
        public bool IsClosed { get; set; }
        public DateTime LastUpdateDate { get; set; }
        public virtual Image Image { get; set; }
        public virtual Category Category { get; set; }
        public virtual IEnumerable<BannerLocalization> BannerLocalizations { get; set; }

    }
}
