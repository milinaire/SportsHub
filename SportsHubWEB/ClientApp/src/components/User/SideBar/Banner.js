import React from "react";
import style from './SideBar.module.css'

const Banner = (props) => {
  return (
    <div className={style.bannerWrapper}>
      <div className={style.bannerBackground} style={{backgroundImage: `url(${props.image})`}}>
        {props.category && <div className={style.bannerCategory}>
          {props.category}
        </div>}
        {props.headline && <div className={style.bannerHeadline}>
          {props.headline}
        </div>}
      </div>

    </div>
  )
}
export default Banner;