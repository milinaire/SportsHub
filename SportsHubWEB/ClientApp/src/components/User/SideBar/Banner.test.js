import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter} from 'react-router-dom';
import Banner from "./Banner";
import renderer from 'react-test-renderer';

describe('my beverage', () => {
  test('renders without crashing', async () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Banner/>
      </MemoryRouter>, div);

  })
  test('',()=>{
    const tree = renderer
      .create(<Banner/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})