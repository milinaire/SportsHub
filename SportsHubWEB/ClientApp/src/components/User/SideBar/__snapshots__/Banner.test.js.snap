// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my beverage  1`] = `
<div
  className="bannerWrapper"
>
  <div
    className="bannerBackground"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
</div>
`;
